tar.go

```diff
@@ -0,0 +1,154 @@
+package archiver
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func init() {
+	RegisterFormat("Tar", t{})
+}
+
+type t struct{}
+
+func (t) Match(filename string) bool {
+	// TODO: read file header to identify the format
```

Comment:
Yes. The original intention here is that we'd better identify the format according to the file header instead of the extension of file name which may be fake. For creating archives, maybe there is no need to use `Match` for checking (a crazy user may create a tarball named `crazy.zip` if he wants...).